commerce-backend-ui-1/web-src/src/components/MainPage.js

```diff
@@ -9,36 +9,75 @@ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTA
 OF ANY KIND, either express or implied. See the License for the specific language
 governing permissions and limitations under the License.
 */
-import { Item, TabList, TabPanels, Tabs } from '@adobe/react-spectrum';
-import { useState } from 'react';
+import { Flex, Item, ProgressCircle, TabList, TabPanels, Tabs, View } from '@adobe/react-spectrum';
+import { attach } from '@adobe/uix-guest';
+import { useEffect, useState } from 'react';
 import { TaxClassesPage } from './TaxClassesPage';
+import { TAX_EXTENSION_ID } from '../constants/extension';
 
 export const MainPage = (props) => {
   const [selectedTab, setSelectedTab] = useState('1');
+  const [imsToken, setImsToken] = useState(null);
+  const [imsOrgId, setImsOrgId] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
 
   const onSelectionTabChange = (selectedTabKey) => {
     setSelectedTab(selectedTabKey);
   };
 
+  useEffect(() => {
+    // Load IMS token for calling require-adobe-auth: true actions
+    const loadImsInfo = async () => {
+      try {
+        if (props.ims?.token) {
+          // When running inside Experience Cloud Shell, IMS token and orgId can be accessed via props.ims.
+          setImsToken(props.ims.token);
+          setImsOrgId(props.ims.org);
+        } else {
+          // Commerce PaaS requires Admin UI SDK 3.0+ to access IMS info via sharedContext.
+          // See https://developer.adobe.com/commerce/extensibility/admin-ui-sdk/extension-points/#shared-contexts
+          const guestConnection = await attach({ id: TAX_EXTENSION_ID });
```

Comment:
I see the main difference from the previous sample is that it retrieves the connection directly instead of getting it from the registration.
https://magento.slack.com/archives/C02NVJGJJK1/p1747329285663489

I tried to make the code sample work for the following cases:
1. Admin UI SDK 3.0 with shared contexts
2. Experience Cloud Shell
3. Where authentication is turned off